packages/go_router/test/go_router_test.dart

```diff
@@ -4863,6 +4863,29 @@ void main() {
       expect(statefulWidgetKeyB.currentState?.counter, equals(1));
     });
   });
+
+  ///Regression tests for https://github.com/flutter/flutter/issues/132557
+  group('overridePlatformDefaultLocation', () {
+    test('No initial location provided', () {
```

Comment:
This test only test for the assert, can you write a test that mocks the initial route from platform?

Reply:
I will be writing test to emulate initial route from platform, but I'd probably need help to implement it. I'll keep you posted.

Reply:
@chunhtai I was unable to find a way to set platform initalroute pragmatically for testing. Requesting help...

Reply:
looks like there isn't an easy way. I think you can test it like this.
```dart
final String platformRoute = WidgetsBinding.instance.platformDispatcher.defaultRouteName;
final String expectedInitialRoute = '/abc';
// Sanity check.
expect(platformRoute != expectedInitialRoute, isTrue)
final GoRouter router = GoRouter(overridePlatformDefaultLocation: true, initialRoute: '/abc', ...);
expect(router.routeInformationProvider.value.location,  expectedInitialRoute);
```

Reply:
This seems not addressed yet?

Reply:
Added new commits.